lib/ddtrace/utils.rb

```diff
@@ -1,9 +1,21 @@
+require 'thread'
+
 module Datadog
-  # TODO[manu]: write docs
+  # Utils contains low-level utilities, typically to provide pseudo-random trace IDs.
   module Utils
+    # We use a custom random number generator because we want no interference
+    # with the default one. Using the default prng, we could break code that
+    # would rely on srand/rand sequences.
+    @rnd = Random.new
+    # We want to garantee all calls are sequential else, in some edge cases,
+    # the same value could be return twice, ending in trace collision.
```

Comment:
I'm not 100.00% sure of this, could be worth a check, but it's hard to prove that you have a collision. I wrote that initially because I had some collisions, but the collisions were coming from elsewhere. I still do think it's relevant, but at that point I could also remove it. Taking any suggestion.